resources/assets/lib/components/report-form.tsx

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { Modal } from 'components/modal';
+import { SelectOptions } from 'components/select-options';
+import { isEmpty } from 'lodash';
+import * as _ from 'lodash';
+import * as React from 'react';
+
+const bn = 'report-form';
+const maxLength = 2000;
+
+interface Props {
+  completed: boolean;
+  disabled: boolean;
+  onClose: () => void;
+  onSubmit: ({comments}: {comments: string}) => void;
+  title: string;
+  visible: boolean;
+  visibleOptions?: string[];
```

Comment:
`visibleOptions` should be required and `defaultProps` declared for the component